Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -651,6 +675,12 @@ static void webkitWebViewBaseRealize(GtkWidget* widget)
     webkitWebViewBaseUpdateDisplayID(webView, monitor);
 #endif
 
+#if GTK_CHECK_VERSION(4, 12, 0)
+    priv->watchedScaleSurface = gtk_native_get_surface(gtk_widget_get_native(widget));
+    priv->watchedScaleSurfaceSignalId = g_signal_connect_swapped(priv->watchedScaleSurface, "notify::scale", G_CALLBACK(deviceScaleFactorChanged), webView);
```

Comment:
Why do we need to keep a pointer to the surface? The changed callback is always getting the surface from the widget.

Reply:
We don't probably need the the singla id either if we use `g_signal_connect_object()`.